runelite-client/src/main/java/net/runelite/client/ui/overlay/components/LineComponent.java

```diff
@@ -0,0 +1,169 @@
+
+		final FontMetrics metrics = graphics.getFontMetrics();
+		int x = 0;
+		int y = 0;
+		final int leftFullWidth = getLineWidth(left, metrics);
+		final int rightFullWidth = getLineWidth(right, metrics);
+
+		if (preferredSize.width < leftFullWidth + rightFullWidth)
+		{
+			int leftSmallWidth = preferredSize.width;
+			int rightSmallWidth = 0;
+
+			if (!Strings.isNullOrEmpty(right))
+			{
+				rightSmallWidth = (preferredSize.width / 3);
+				leftSmallWidth -= rightSmallWidth;
+			}
+
+			final String[] leftSplitLines = lineBreakText(left, leftSmallWidth, metrics);
+			final String[] rightSplitLines = lineBreakText(right, rightSmallWidth, metrics);
+
+			int lineCount = Math.max(leftSplitLines.length, rightSplitLines.length);
+
+			for (int i = 0; i < lineCount; i++)
+			{
+				String leftText = "";
+				String rightText = "";
+
+				if (i < leftSplitLines.length)
+				{
+					leftText = leftSplitLines[i];
+				}
+
+				if (i < rightSplitLines.length)
+				{
+					rightText = rightSplitLines[i];
+				}
+
+				final TextComponent leftLineComponent = new TextComponent();
+				leftLineComponent.setPosition(new Point(x, y));
+				leftLineComponent.setText(leftText);
+				leftLineComponent.setColor(leftColor);
+				leftLineComponent.render(graphics);
+
+				final TextComponent rightLineComponent = new TextComponent();
+				rightLineComponent.setPosition(new Point(x + leftSmallWidth + rightFullWidth - getLineWidth(rightText, metrics), y));
+				rightLineComponent.setText(rightText);
+				rightLineComponent.setColor(rightColor);
+				rightLineComponent.render(graphics);
+				y += metrics.getHeight() + SEPARATOR;
+			}
+
+			return new Dimension(preferredSize.width, y);
+		}
+
+		final TextComponent leftLineComponent = new TextComponent();
+		leftLineComponent.setPosition(new Point(x, y));
+		leftLineComponent.setText(left);
+		leftLineComponent.setColor(leftColor);
+		leftLineComponent.render(graphics);
+
+		final TextComponent rightLineComponent = new TextComponent();
+		rightLineComponent.setPosition(new Point(x + preferredSize.width - getLineWidth(right, metrics), y));
+		rightLineComponent.setText(right);
+		rightLineComponent.setColor(rightColor);
+		rightLineComponent.render(graphics);
+		y += metrics.getHeight() + SEPARATOR;
+
+		return new Dimension(preferredSize.width, y);
+	}
+
+	private static int getLineWidth(final String line, final FontMetrics metrics)
+	{
+		return metrics.stringWidth(TextComponent.textWithoutColTags(line));
+	}
+
+	private static String[] lineBreakText(String text, int maxWidth, FontMetrics metrics)
+	{
+		final String[] words = text.split(" ");
+
+		if (words.length == 0)
+		{
+			return new String[0];
+		}
+
+		final StringBuilder wrapped = new StringBuilder(words[0]);
+		int spaceLeft = maxWidth - metrics.stringWidth(wrapped.toString());
+
+		for (int i = 1; i < words.length; i++)
+		{
+			final String word = words[i];
+			final int wordLen = metrics.stringWidth(word);
+
+			if (wordLen + 3 > spaceLeft)
```

Comment:
Where does `3` come from? 
You could probably measure a `spaceWidth = metrics.stringWidth(" ");` and use that instead.